scipy/signal/_peak_finding.py

```diff
@@ -277,7 +350,19 @@ def peak_prominences(x, peaks, wlen=None):
     Raises
     ------
     ValueError
-        If an index in `peaks` does not point to a local maximum in `x`.
+        If a value in `peaks` is an invalid index for `x`.
+
+    Warns
+    -----
+    PeakPropertyWarning
+        For indices in `peaks` that don't point to valid local maxima in `x`
+        the returned prominence will be 0 and this warning is raised. This
+        also happens if `wlen` is smaller than the plateau size of a peak.
+
+    Warnings
```

Comment:
I don't know if this is called Notes or Warnings in this context. But also I don't have any strong opinion between the two

Reply:
This was originally inside the directive `.. warnings::` in the Notes section. However [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#sections) explicitly supports this section as the place for warnings. So I felt this would be more inline with the documentation style chosen by the  SciPy community.
There was already some discussion on whether this is should be a warning or a note. If you are interested you can look at gh-8705 (highlight https://github.com/scipy/scipy/pull/8705#issuecomment-380586080) for why a warning was chosen.